sklearn/decomposition/_dict_learning.py

```diff
@@ -865,6 +902,98 @@ def dict_learning_online(
     SparsePCA
     MiniBatchSparsePCA
     """
+    deps = (return_n_iter, return_inner_stats, iter_offset, inner_stats)
+    if max_iter is not None and not all(arg == "deprecated" for arg in deps):
+        raise ValueError(
+            "the following args are incompatible with 'max_iter': "
+            "return_n_iter, return_inner_stats, iter_offset, inner_stats"
+        )
+
+    if iter_offset != "deprecated":
+        warnings.warn(
+            "'iter_offset' is deprecated in version 1.1 and "
+            "will be removed in version 1.3.",
+            FutureWarning,
+        )
+    else:
+        iter_offset = 0
+
+    if return_inner_stats != "deprecated":
+        warnings.warn(
+            "'return_inner_stats' is deprecated in version 1.1 and "
+            "will be removed in version 1.3.",
```

Comment:
The deprecation message should state what is going to be the future behavior: never return the inner stat.